mapping_project/lib/saved_locs.dart

```diff
@@ -66,61 +72,131 @@ class _SavedLocsState extends State<SavedLocs> {
+                decoration: InputDecoration(labelText: "Room")),
+            actions: <Widget>[
+              MaterialButton(
+                //Creates submit button because it looks neater
+                elevation: 5.0,
+                child: Text('Submit'),
+                onPressed: () {
+                  Navigator.of(context).pop(customController.text
+                      .toString()); //Pop! The user's inputs are taken and relayed (in a list<string>) to where the function was called from
+                },
+              )
+            ],
+          );
+        });
+  }
+
+  Future<List<String>> createPopupBox2(
       BuildContext context,
+      bool addorremove,
       bool
-          addorremove) //Creates popup box which allows user to enter the name of the location
+          name) //Creates popup box which allows user to enter the name of the location
   {
     TextEditingController customController = TextEditingController();
+    TextEditingController customController2 = TextEditingController();
+    List<String> lstr = new List<String>();
 
     String adder =
         "remove"; //Bit of cosmetics - if user is removing location, says remove etc...
     if (addorremove == true) {
       adder = "add";
     }
 
+    String text2 = "location";
+    if (name) {
+      text2 = "description";
+    }
+
     return showDialog(
         context: context,
         builder: (context) {
           return AlertDialog(
-            title: Text("Enter location to $adder:"),
-            content: TextField(
-              controller:
-                  customController, //CustomController allows me to return Future<String> for this and so I can get the user's input
-            ),
+            title: Text("Enter $text2 to $adder:"),
+            content: SingleChildScrollView(
+                child: ListBody(children: <Widget>[
+              TextField(
+                controller: customController,
+                decoration: InputDecoration(labelText: "Room"),
+              ),
+              TextField(
+                controller: customController2,
+                decoration: InputDecoration(labelText: "Description"),
+              )
+            ])),
             actions: <Widget>[
               MaterialButton(
                 //Creates submit button because it looks neater
                 elevation: 5.0,
                 child: Text('Submit'),
                 onPressed: () {
-                  Navigator.of(context).pop(customController.text
-                      .toString()); //Pop! The user's input is taken and relayed to where the function was called from
+                  lstr.add(customController.text.toString());
+                  lstr.add(customController2.text.toString());
+                  Navigator.of(context).pop(
+                      lstr); //Pop! The user's inputs are taken and relayed (in a list<string>) to where the function was called from
                 },
```

Comment:
Correct me if I'm wrong - but I can't see anything here for reading from and writing to the savedLocs file?